plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/MergeRollbackAction.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import io.trino.spi.TrinoException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.trino.plugin.jdbc.JdbcErrorCode.JDBC_ERROR;
+
+public class MergeRollbackAction
```

Comment:
If you go with my suggestion abover you may just drop this class and replace with method which executes all rollback actions on list